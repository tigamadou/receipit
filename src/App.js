function App() {
  return (
    <div className="App">
      <header className="App-header">
        My App
      </header>
    </div>
  );
}

export default App;
