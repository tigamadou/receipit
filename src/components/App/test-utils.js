import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import HeaderReducer from '../../redux/reducers/HeaderRecuder/HeaderReducer';

function render(
  ui,
  {
    initialState,
    store = createStore(HeaderReducer, initialState),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { render };
