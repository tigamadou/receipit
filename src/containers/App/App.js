import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import AppRouter from '../../routers/AppRouter';
import Header from '../../components/Header/Header';

const App = ({ header }) => (
  <div className="app">
    <Header data={header} />
    <div className="wrapper">
      <AppRouter />
    </div>
  </div>
);
App.propTypes = {
  header: PropTypes.instanceOf(Object),
};
App.defaultProps = {
  header: null,
};

const mapStateToProps = (state) => ({
  header: state.header,
});

export default connect(mapStateToProps, null)(App);
