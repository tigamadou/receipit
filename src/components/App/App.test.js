import React from 'react';
import {
  BrowserRouter as Router,
} from 'react-router-dom';
import { render, screen } from './test-utils';
import App from './App';

describe('App connected component', () => {
  it('should render with given state from Redux store', () => {
    render(<Router><App /></Router>, { initialState: { header: { title: 'Recipely', back: false } } });
    expect(screen.getByText('Recipely')).toBeInTheDocument();
  });
});
