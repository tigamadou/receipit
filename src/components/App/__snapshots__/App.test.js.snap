// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App connected component should match with snapshot 1`] = `
<div
  className="app"
>
  <header
    className="header dark"
  >
    <div
      className="title white is_sm"
    >
      <h1>
        Recipy
      </h1>
    </div>
  </header>
  <div
    className="wrapper"
  >
    <div
      className="container"
      data-testid="home"
    >
      <div
        className="cards"
      />
    </div>
  </div>
</div>
`;
